### PR TITLE
Drive traffic to credential chains Learn doc

### LINKED ISSUE #33356: [BUG]
**Describe the bug**
The Azure Identity [README](https://github.com/Azure/azure-sdk-for-java/blob/74558d770959d283454bdb0842e70e9b43243908/sdk/identity/azure-identity/README.md#L126) contains a broken link in [line 126](https://github.com/Azure/azure-sdk-for-java/blob/74558d770959d283454bdb0842e70e9b43243908/sdk/identity/azure-identity/README.md?plain=1#L126) where it says "Configure DefaultAzureCredential".

***Exception or Stack Trace***
N/A

**To Reproduce**
Click on the link.

***Code Snippet***
N/A

**Expected behavior**
The link should point to a guide on how to use the credential type in question.

**Screenshots**
N/A

**Setup (please complete the following information):**
 - OS: All?
 - IDE: N/A
 - Library/Libraries: com.azure:azure-identity
 - Java version: N/A
 - App Server/Environment: N/A
 - Frameworks: N/A

If you suspect a dependency version mismatch (e.g. you see `NoClassDefFoundError`, `NoSuchMethodError` or similar), please check out [Troubleshoot dependency version conflict article](https://aka.ms/azsdk/java/dependency/troubleshoot) first. If it doesn't provide solution for the problem, please provide:
- verbose dependency tree (`mvn dependency:tree -Dverbose`)
- exception message, full stack trace, and any available logs

**Additional context**
N/A

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [X] Bug Description Added
- [X] Repro Steps Added
- [X] Setup information Added

### PR DESCRIPTION
**Pull request by scottaddie**
Remove content from the Azure Identity README that now exists in the credential chains doc on Learn. Also update and simplify API docs for `ChainedTokenCredential` and `DefaultAzureCredential`.

Fixes #33356

### CODE REVIEW

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredentialBuilder.java [hunk 1]
```diff
@@ -11,11 +11,7 @@
 import java.util.Deque;
 
 /**
- * Fluent credential builder for instantiating a {@link ChainedTokenCredential}.
- *
- * The {@link ChainedTokenCredential} is a convenience credential that allows users to chain together a set of
- * TokenCredential together. The credential executes each credential in the chain sequentially and returns the token
- * from the first credential in the chain that successfully authenticates.
+ * <p>Fluent credential builder for instantiating {@link ChainedTokenCredential}.</p>
```
Comment (billwert) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredentialBuilder.java:14]: Can we link the doc here as you did for DAC?

**Review by billwert — APPROVED** [review]

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredentialBuilder.java (hunk 1 above)
Comment (scottaddie) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredentialBuilder.java:14]: My thought was we'd link to the doc from the credential itself, not the builder. Does that make sense?

Comment (billwert) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredentialBuilder.java:14]: Ah, sure. I thought DAC had a link in both places, my mistake.

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredential.java [hunk 2]
```diff
@@ -9,34 +9,24 @@
...
- * attempt to authenticate via the following mechanisms in order.</p>
+ * <p>DefaultAzureCredential simplifies authentication while developing apps that deploy to Azure by combining credentials
+ * used in Azure hosting environments with credentials used in local development. In production, it's better to use something
+ * else. For more information, see <a href="https://aka.ms/azsdk/java/identity/credential-chains#usage-guidance-for-defaultazurecredential">Usage guidance for DefaultAzureCredential</a>.
  *
- * <img src="doc-files/DefaultAzureCredentialAuthFlow.png" alt="">
+ * <p>Attempts to authenticate with each of these credentials, in the following order, stopping when one provides a token:</p>
  *
  * <ol>
- * <li>{@link EnvironmentCredential} - The DefaultAzureCredential will read account information specified via
```
Comment (g2vinay) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredential.java:20]: This does add an extra layer for dev to unwrap from IDE to learn about the usage.
The idea previously was to provide the conceptual info needed to use the credential within the Javadocs..

**Review by g2vinay — APPROVED** [review]

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredential.java (hunk 2 above)
Comment (scottaddie) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredential.java:20]: I'm fine with that. It forces the customer to read the linked doc.

**Pull request merged**